emails/models.py

```diff
@@ -253,6 +254,54 @@ def add_subdomain(self, subdomain):
         RegisteredSubdomain.objects.create(subdomain_hash=hash_subdomain(subdomain))
         return subdomain
 
+    def update_abuse_metric(self, address_created=False, replied=False):
+        #  TODO: this should be wrapped in atomic to ensure race conditions are properly handled
+        # look for abuse metrics created on the same UTC date, regardless of time.
+        midnight_utc_today = datetime.combine(
+            datetime.now(timezone.utc).date(), datetime.min.time()
+        ).astimezone(timezone.utc)
+        midnight_utc_tomorow = datetime.combine(
+            datetime.now(timezone.utc).date(), datetime.min.time()
+        ).astimezone(timezone.utc) + timedelta(days=1)
+        abuse_metric = self.user.abusemetrics_set.filter(
+            first_recorded__gte=midnight_utc_today, first_recorded__lt=midnight_utc_tomorow
+        ).first()
+        if not abuse_metric:
+            abuse_metric = AbuseMetrics.objects.create(user=self.user)
```

Comment:
While we're here, could we clean up old `AbuseMetrics` records we don't need anymore? E.g.,
```
AbuseMetrics.objects.delete(first_recorded__lt=midnight_utc_today)
```